hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -149,7 +152,8 @@ public abstract class HoodieBackedTableMetadataWriter<I, O> implements HoodieTab
   private static final String RECORD_KEY_FIELD_NAME = HoodieMetadataPayload.KEY_FIELD_NAME;
 
   // tracks the list of MDT partitions which can write to metadata table in a streaming manner.
-  private static final List<MetadataPartitionType> STREAMING_WRITES_SUPPORTED_PARTITIONS = Arrays.asList(RECORD_INDEX);
+  private static final List<MetadataPartitionType> STREAMING_WRITES_SUPPORTED_PARTITIONS = Arrays.asList(RECORD_INDEX, SECONDARY_INDEX);
+  private static final List<MetadataPartitionType> STREAMING_WRITES_SUPPORTED_PARTITION_PREFIXES = Arrays.asList(SECONDARY_INDEX);
```

Comment:
Note to Reviewer: 
Certain partitions in MDT are one on one mapped to the MetadataPartitionType. but its not always the case. 
For eg, in case of secondary index, we generally do `startsWtith` and not really exact match, since we could have N no of sec indexes built.

Reply:
can we enhance tests that we wrote for the patch where we added APIs and impl for the streaming metadata writes.